Simplenote/SPAppDelegate+Extensions.swift

```diff
@@ -399,6 +399,24 @@ extension SPAppDelegate {
     }
 }
 
+// MARK: - Account Deletion
+//
+extension SPAppDelegate {
+    @objc
+    func authenticateSimperiumIfNeeded() {
+        guard let deletionController = accountDeletionController else {
+            return
+        }
+
+        if deletionController.deletionTokenHasExpired {
+            self.accountDeletionController = nil
+            return
+        }
+
+        simperium.authenticateIfNecessary()
+    }
```

Comment:
Unfortunately this won't work as expected and can end up crashing the app.
When accessing objc code via swift it will go via property getter, which will create a new `accountDeletionController`.

Imagine the following flow:
* Launch the app
* Put into background
* Put into foreground, `authenticateSimperiumIfNeeded` is invoked and `accountDeletionController` is created. `accountDeletionRequestDate` is set to the current date.
* Because the token is not expired, we call `simperium.authenticateIfNecessary()` every time the app is opened (and we didn't initiate account deletion)
* If we request account deletion within a day, the expiration date won't be correct as it was already set when `accountDeletionController` was created
* If we request account deletion after a day has passed (considering the app wasn't killed), the app will crash as `accountDeletionController` will be nil.